docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+parameters.
+
+The next sections describe the witness server, how to set-up a sidechain, error
+handling, and trust assumptions.
+
+Finally, a detailed reference for the new ledger objects and transactions are
+presented as a reference.
+
+## Nomenclature
+
+Cross-chain transfer: A transaction that moves assets from the mainchain to the
+sidechain, or returns those assets from the sidechain to the mainchain
+
+Cross-chain sequence number: A ledger object used to prove ownership of the
+funds moved in a cross-chain transfer.
+
+Door accounts: The account on the mainchain that is used to put assets into
+trust, or the account on the sidechain used to issue wrapped assets. The name
+comes from the idea that a door is used to move from one room to another and a
+door account is used to move from one chain to another.
+
+Mainchain: Where the assets originate and are put into trust.
+
+Sidechain: Where the assets from the mainchain are wrapped.
+
+Witness server: A server that listens for transactions on the main and side
+chains and signs attestations used to prove that certain events happened on each
+chain.
+
+## List of new transactions
+
+SidechainCreate
+SidechainXChainSeqNumCreate
+SidechainXChainTransfer
+SidechainXChainClaim
+SidechainXChainAccountCreate
+SidechainAddWitness
+SidechainModify
+
+## List of new ledger objects
+
+Sidechain
+CrosschainSeqNum
+XChainAccountCreateState
+XChainTransferRewardState
+
+## Cross-chain transfers overview
+
+A cross-chain transfer moves assets from the mainchain to the sidechain or
+returns those assets from the sidechain back to the mainchain. Cross-chain
+transfers need a couple primitives:
+
+1) Put assets into trust on the mainchain.
+
+2) Issue wrapped assets on the sidechain.
+
+3) Return or destroy the wrapped assets on the sidechain.
+
+4) On the sidechain, prove that assets were put into trust on the mainchain.
+
+5) On the mainchain, prove that assets were returned or destroyed.
+
+6) A way to prevent assets from being wrapped multiple times (prevent
+transaction replay). The proofs that certain events happened on the different
+chains are public and can be submitted multiple times. This can only wrap (or
+unlock) assets once.
+
+In this implementation, a regular XRP ledger account is used to put assets into
+trust on the mainchain, and a regular XRP ledger account is used to issue assets
+on the sidechain. These accounts will have their master keys disabled and funds
+will move from these accounts through a new set of transactions specifically
+meant for cross-chain transfers. A special set of "witness servers" are used to
+prove that assets were put into trust on the mainchain or returned on the
+sidechain. A new ledger object called a "cross-chain sequence number" is used to
+prevent transaction replay.
+
+A source chain is the chain where the cross-chain transfer start - by either
+putting assets into trust on the mainchain or returning wrapped assets on the
```

Comment:
@seelabs, have we tested the same flow with sidechain -> mainchain cross chain transfer where the wrapped asset is burned. In this case I assume the seq number is checked out on the mainchain and claim trx happens on the mainchain with the difference is the asset gets burned on sidechain as opposed to being locked for mainchain -> sidechain cross chain transfer.

Reply:
Yes, it works as you described, and I have tested this.

Reply:
sounds good